lstchain/image/muon/muon_analysis.py

```diff
@@ -131,90 +129,135 @@ def analyze_muon_event(event_id, image, geom, equivalent_focal_length,
+        muonintensityoutput.ring_completeness = ring_completeness(
+            x[dist_ringwidth_mask], y[dist_ringwidth_mask], image[dist_ringwidth_mask],
+            muonringparam.ring_radius,
+            muonringparam.ring_center_x,
+            muonringparam.ring_center_y,
+            threshold=30,
+            bins=30)
+
+        pix_ringwidth_im = image * dist_ringwidth_mask
+        idx_ringwidth = np.nonzero(pix_ringwidth_im)
+        muonintensityoutput.ring_pix_completeness = npix_above_threshold(
+            pix_ringwidth_im[idx_ringwidth], tailcuts[0]) / len(
+                pix_ring[idx_ringwidth])
+
+    else:
+            muonintensityoutput = MuonIntensityParameter()
+            # Set default values for cases in which the muon intensity fit (with MuonLineIntegrate) is not done:
+            muonintensityoutput.ring_width = np.nan*u.deg
+            muonintensityoutput.impact_parameter_pos_x = np.nan*u.m
+            muonintensityoutput.impact_parameter_pos_y = np.nan*u.m
+            muonintensityoutput.impact_parameter = np.nan*u.m
+            muonintensityoutput.ring_pix_completeness = np.nan
+            muonintensityoutput.ring_completeness = np.nan
+
 
-    muonintensityoutput = ctel.fit_muon(muonringparam.ring_center_x,
-                                    muonringparam.ring_center_y,
-                                    muonringparam.ring_radius,
-                                    x[dist_mask], y[dist_mask],
-                                    image[dist_mask])
     muonintensityoutput.mask = dist_mask
-    idx_ring = np.nonzero(pix_ring)
-    muonintensityoutput.ring_completeness = ring_completeness(
-                    x[idx_ring], y[idx_ring], pix_ring[idx_ring],
-                    muonringparam.ring_radius,
-                    muonringparam.ring_center_x,
-                    muonringparam.ring_center_y,
-                    threshold=30,
-                    bins=30)
     muonintensityoutput.ring_size = np.sum(pix_ring)
-    size_outside_ring = np.sum(pix_out_ring * clean_mask)
-    dist_ringwidth_mask = np.abs(dist - muonringparam.ring_radius
-                                             ) < (muonintensityoutput.ring_width)
-    pix_ringwidth_im = image * dist_ringwidth_mask
-    idx_ringwidth = np.nonzero(pix_ringwidth_im)
-
-    muonintensityoutput.ring_pix_completeness = npix_above_threshold(
-                    pix_ringwidth_im[idx_ringwidth], tailcuts[0]) / len(
-                    pix_ring[idx_ringwidth])
-
-    print("Impact parameter = %s"
-                             "ring_width=%s, ring radius=%s, ring completeness=%s"% (
-                             muonintensityoutput.impact_parameter,
-                             muonintensityoutput.ring_width,
-                             muonringparam.ring_radius,
-                             muonintensityoutput.ring_completeness))
+    size_outside_ring = np.sum(pix_outside_ring * clean_mask)
```

Comment:
be careful here, because
`ring_size ` is calculated _without_ applying cleaning
`size_outside_ring ` is calculated applying cleaning

therefore the total 

Also, I suggest we move `size` -> `intensity` to be compatible with the rest of the naming. `ring_size` should also be renamed, but this is a ctapipe inherited variable...

Reply:
sorry, the complete answer was:
"therefore the total size of the _cleaned_ or _uncleaned_ image will not be equal to `ring_size + size_outside_ring"

ok for the `size` renaming

Reply:
> sorry, the complete answer was:
> "therefore the total size of the _cleaned_ or _uncleaned_ image will not be equal to `ring_size + size_outside_ring"
> 
Ok, that is clear, but I see no problem with it.